dask/dataframe/io/sql.py

```
@@ -211,3 +213,173 @@ def _read_sql_chunk(q, uri, meta, engine_kwargs=None, **kwargs):
+    uri : string
+        Full sqlalchemy URI for the database connection
+    schema : str, optional
+        Specify the schema (if database flavor supports this). If None, use
+        default schema.
+    if_exists : {'fail', 'replace', 'append'}, default 'fail'
+        How to behave if the table already exists.
+
+        * fail: Raise a ValueError.
+        * replace: Drop the table before inserting new values.
+        * append: Insert new values to the existing table.
+
+    index : bool, default True
+        Write DataFrame index as a column. Uses `index_label` as the column
+        name in the table.
+    index_label : str or sequence, default None
+        Column label for index column(s). If None is given (default) and
+        `index` is True, then the index names are used.
+        A sequence should be given if the DataFrame uses MultiIndex.
+    chunksize : int, optional
+        Specify the number of rows in each batch to be written at a time.
+        By default, all rows will be written at once.
+    dtype : dict or scalar, optional
+        Specifying the datatype for columns. If a dictionary is used, the
+        keys should be the column names and the values should be the
+        SQLAlchemy types or strings for the sqlite3 legacy mode. If a
+        scalar is provided, it will be applied to all columns.
+    compute : bool, default True
+        When true, call dask.compute and perform the load into SQL; otherwise, return a Dask object (or array of
+        per-block objects when parallel=True)
+    parallel : bool, default False
+        When true, have each block append itself to the DB table concurrently. This can result in DB rows being in a
+        different order than the source DataFrame's corresponding rows. When false, load each block into the SQL DB in
+        sequence.
+    kwargs : dict
+        If Pandas version is >= 0.24.0, a "method" value may be passed here; no other kwargs are allowed.
+
+        .. versionadded:: 0.24.0
```

Just accept a `method` keyword here regardless of the version. Then if pandas <= 0.24.0 you can raise an exception if the default is not provided.

Then we don't have to validate that we aren't silently discarding arguments.